### ISSUE TITLE
Rule proposal: No mutiple array check in logical expression

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#2366]
### Description

Use `.some()` or `.every()` check on one array can simplify to use one call.

### Fail

```js
if (
	array.some(element => element.foo === 1)
	|| array.some(element => element.bar === 2)
) {}
```

```js
if (
	array.every(element => element.foo === 1)
	&& array.every(element => element.bar === 2)
) {}
```

### Pass

```js
if (
	array.some(element => element.foo === 1 || element.bar === 2)
) {}
```

```js
if (
	array.every(element => element.foo === 1 && element.bar === 2)
) {}
```

### Proposed rule name

Not sure about the name yet

**Comment by silverwind** [comment]
I think it could als handle `.find` and its variants exactly the same as it would `.some`.

**Comment by sindresorhus** [comment]
Rule name: `no-multiple-array-checks-in-logical-expressions`?

**Comment by fisker** [comment]
> Inside boolean context, I think it could als handle `.find` and its variants exactly the same as it would `.some`.

Did you mean

```js
if (
	array.find(element => element.foo === 1)
	&& array.find(element => element.bar === 2)
) {}
```

?

[prefer-array-some](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-array-some.md) should already handled?

**Comment by sindresorhus** [comment]
Accepted
